WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES/WINGSTOP_Air_Brake_Flight_Computer_FULL_BRAKES.ino

```diff
@@ -67,30 +67,63 @@ READ ME
 
 //VARIABLES
 
-  int c = 0; //COUNTER
-  float x1 = 0, x2 = 0, vel = 0, acc = 0, acc_avg = 0, apo = 0, apo_act = 0, init_pressure = 0, init_altitude = 0, delta_t = 0; //DATA VARIABLES (decimal)
-  unsigned long t1 = 0, t2 = 0; //TIMER VARIABLES (integers)
+  int counter_mco = 0; //COUNTER for Main Engine Cutoff detection
+  int counter_apogee = 0; //COUNTER for Apogee detection
+  int counter_landed = 0; //COUNTER for Landed Detection
+  
+  float x_previous = 0; // previous position
+  float x_current = 0; //current position
+  float vel = 0; //velocity
+  float acc = 0; //acceleration
+  float acc_avg = 0; //acceleration variable for averaging data
+  //float apo = 0; //predicted apogee
+  float init_pressure = 0; //ground pressure reading in HPa
+  float init_altitude = 0; //ground altitude reading in meters
+  float delta_t = 0; //time between iterations in milliseconds
+  float target_acc = 0; //acceleration needed to hit target altitude
+  float distance_to_target = 0; //distance to target apogee
+  float brake_position = 0; //position to set brakes to (0 for closed, 100 for open full)
+
+  //PID values:
+  float P_gain = 2; //gain value for p controller
+  
+  
+  unsigned long t_previous = 0;  //previous clock time in milliseconds
+  unsigned long t_current = 0;  //current clock time in milliseconds
   
-  //brakes servo RANGE
+  //brakes servo RANGE and Velocity Limit
   #define brakes_closed  625
   #define brakes_open  1580
+  #define max_brake_velocity 500 //must be under this velocity in [m/s] to deploy brakes
   
   //DEFINE PIN NUMBERS
   #define BUZZER_PIN  15
   #define LED_PIN  5
   #define SERVO_PIN  3
+
+
+
   
 //**************SET TARGET ALTITUDE HERE*************************
-  float target_altitude = 100; //target altitude above ground in meters
+  #define TARGET_ALTITUDE (1) //target altitude above ground in meters
 
 // *********SET SEA LEVEL PRESSURE HERE***************
   #define SEALEVELPRESSURE_HPA (1016.00)//set according to location and date
   
 // *********SET FREQUENCY HERE IN HZ***************
-  #define hz (5)//hz
+  #define hz (10)//hz
+
+// *********GRAVITY CONSTANT***************
+  #define G (9.81)// m/(s^2)
+
+// *********SET TAKEOFF ALTITUDE HERE***************
+  #define TAKEOFF_ALTITUDE  (1) //altitude above the ground in meters that triggers the detect_take_off function
+
+// *********SET ACCELERATION OFFSET CONSTANT HERE***************
+  #define OFFSET (-7)// offset for acceleration data in m/(s^2) for prediction algorithm
+
+
 
-// *********SET GRAVITY CONSTANT HERE***************
-  #define G (6.3)// m/(s^2)
 
 //Brakes Servo Handle
   Servo brakes; 
```

Comment:
This comment is actually for line 99/131 below.  What is the comment next to the `mpu` object mean?